### PR TITLE
refactor: move from io/ioutil to io and os package

### PR DESCRIPTION
**Pull request by Juneezee**
The `io/ioutil` package has been deprecated in Go 1.16 (See https://golang.org/doc/go1.16#ioutil). Since kopia has upgraded to Go 1.17 (#1256), this PR replaces the existing `io/ioutil` functions with their new definitions in `io` and `os` packages.

### CODE REVIEW

File: internal/grpcapi/repository_server.pb.go [hunk 1]
```diff
@@ -7,10 +7,11 @@
 package grpcapi
 
 import (
-	protoreflect "google.golang.org/protobuf/reflect/protoreflect"
-	protoimpl "google.golang.org/protobuf/runtime/protoimpl"
 	reflect "reflect"
 	sync "sync"
+
```
Comment (jkowalski) [inline internal/grpcapi/repository_server.pb.go:12]: revert this change? this file is auto-generated.

File: internal/grpcapi/repository_server_grpc.pb.go [hunk 2]
```diff
@@ -4,6 +4,7 @@ package grpcapi
 
 import (
 	context "context"
+
```
Comment (jkowalski) [inline internal/grpcapi/repository_server_grpc.pb.go:7]: revert this change? this file is auto-generated.

**Review by jkowalski — CHANGES_REQUESTED** [review]

File: internal/logfile/logfile.go [hunk 3]
```diff
@@ -248,31 +247,42 @@ func sweepLogDir(ctx context.Context, dirname string, maxCount int, maxAge time.
 		maxCount = math.MaxInt32
 	}
 
-	entries, err := ioutil.ReadDir(dirname)
+	entries, err := os.ReadDir(dirname)
```
[the remark this answers is not in the document]
Comment (jkowalski) [reply internal/logfile/logfile.go:250]: (as it turns out there are test failures related to this too, see the action logs)

**Comment by jkowalski** [comment]
can you also add linter `forbidigo` rule for "io/ioutil" and silence the remaining warnings so that we don't add new ones?

File: .golangci.yml [hunk 4]
```diff
@@ -6,6 +6,13 @@ linters-settings:
     statements: 60
   forbidigo:
     forbid:
+    - ioutil.Discard # use io.Discard
+    - ioutil.NopCloser # use io.NopCloser
+    - ioutil.ReadAll # use io.ReadAll
+    - ioutil.ReadFile # use os.ReadFile
+    - ioutil.TempDir # use os.MkdirTemp
+    - ioutil.TempFile # use os.CreateTemp
+    - ioutil.WriteFile # use os.WriteFile
```
Comment (Juneezee) [inline .golangci.yml:15]: @jkowalski After we have moved `ioutil.ReadDir` to `os.ReadDir`, we can replace this with [depguard](https://golangci-lint.run/usage/linters/#depguard) to blacklist the `io/ioutil` package completely

**Comment by Juneezee** [comment]
I'm not sure why one of the test case failed for `Make (ubuntu-latest)` but not the others. The full logs is large so I have only copied the part related to the failing test below:

```
2021-10-06T15:30:11.8102105Z     index_blob_manager_v0_test.go:259: err: deleted flag does not match for a1-000030: false want true
2021-10-06T15:30:11.8103109Z         github.com/kopia/kopia/repo/content.verifyFakeContentsWritten
2021-10-06T15:30:11.8104486Z         	/home/runner/work/kopia/kopia/repo/content/index_blob_manager_v0_test.go:449
2021-10-06T15:30:11.8105646Z         github.com/kopia/kopia/repo/content.TestIndexBlobManagerStress.func2
2021-10-06T15:30:11.8107901Z         	/home/runner/work/kopia/kopia/repo/content/index_blob_manager_v0_test.go:213
2021-10-06T15:30:11.8108705Z         golang.org/x/sync/errgroup.(*Group).Go.func1
2021-10-06T15:30:11.8110177Z         	/home/runner/go/pkg/mod/golang.org/x/sync@v0.0.0-20210220032951-036812b2e83c/errgroup/errgroup.go:57
2021-10-06T15:30:11.8110920Z         runtime.goexit
2021-10-06T15:30:11.8111906Z         	/opt/hostedtoolcache/go/1.17.1/x64/src/runtime/asm_amd64.s:1581
2021-10-06T15:30:11.8112548Z         actor[1] error verifying contents
2021-10-06T15:30:11.8113557Z         github.com/kopia/kopia/repo/content.TestIndexBlobManagerStress.func2
2021-10-06T15:30:11.8114752Z         	/home/runner/work/kopia/kopia/repo/content/index_blob_manager_v0_test.go:214
2021-10-06T15:30:11.8115620Z         golang.org/x/sync/errgroup.(*Group).Go.func1
2021-10-06T15:30:11.8117563Z         	/home/runner/go/pkg/mod/golang.org/x/sync@v0.0.0-20210220032951-036812b2e83c/errgroup/errgroup.go:57
2021-10-06T15:30:11.8118367Z         runtime.goexit
2021-10-06T15:30:11.8119053Z         	/opt/hostedtoolcache/go/1.17.1/x64/src/runtime/asm_amd64.s:1581
2021-10-06T15:30:11.8120188Z --- FAIL: TestIndexBlobManagerStress (24.74s)
```
